tests/test_optimize.py

```diff
@@ -747,16 +747,41 @@ def reset(opt):
         opt.reset()
 
 
+@pytest.fixture
```

Comment:
Changed to this structure, as it seems that the previous code left state in between test cases. This made 2 tests fail:

Before: https://github.com/PennyLaneAI/pennylane/runs/3244508688

Reply:
Nice! 💯